nautobot/docs/user-guides/custom-fields.md

```diff
@@ -0,0 +1,344 @@
+
+#### Move to Advanced Tab
+
+When selected, the custom field will appear in the advanced tab of the object detail view instead of the default tab.
+
+### Assignment
+
+#### Content Types
+
+The list of content types to add this custom field to. Only models that subclass the `nautobot.extras.models.customfields.CustomFieldModel` model class can be selected.
+
+### Validation Rules
+
+Validation rules are used for constraining custom fields to specific values.
+
+#### Minimum value
+
+Minimum allowed value for `Integer` fields.
+
+#### Maximum value
+
+Maximum allowed value for `Integer` fields.
+
+#### Validation Regex
+
+Regular expression to enforce on `Text`, `URL`, `Selection` and `Multiple selection` field values. Regex validation is handled by the [python re engine](https://docs.python.org/3/library/re.html) which uses a PCRE or perl-like regular expression syntax. Examples of common regex validations:
+
+Must start with companyname
+> ^companyname
+
+Must end with 5 digit zip code
+> [0-9]{5}$
+
+Must only contain digits
+> ^\d+$
+
+Must be exactly 8 alphanumeric characters
+> ^[0-9a-zA-Z]{8}$
+
+Must be between 8 and 10 alphanumeric characters and underscore
+> ^\w+{8,10}$
+
+Must contain anything that is not whitespace
+> \S
+
+### Custom Field Choices
+
+The choices to be presented for `Selection` and `Multiple selection` custom field types. These are displayed in the order of the weight values supplied with the lowest weight on top. If regex validation is being used, these choices must match the regular expression.
+
+![Custom Field Choices](./images/custom-fields/custom_field_choices.png)
+
+![Custom Field Select](./images/custom-fields/custom_field_select.png)
+
+## Editing Custom Fields
+
+Since automatic provisioning is only performed when a custom field's content types change, some changes made to existing custom fields are not reflected on the associated objects automatically. Some examples of cases where this might cause unexpected behavior are changes to the `required`, `default` and validation fields. If a custom field is created with `required=False` and then later changed to `required=True`, all of the associated objects will fail validation the next time they're saved unless the custom field is updated with a valid value.
```

Comment:
```suggestion
Since automatic provisioning is only performed when a custom field's content types change, some changes made to existing custom fields are not reflected on the associated objects automatically. Some examples of cases where this might cause unexpected behavior are changes to the `required`, `default` and validation fields.

If a custom field is created with `required=False` and then later changed to `required=True`, all of the associated objects that do not yet have a value for this field will fail validation the next time they're saved unless updated with a valid value. Similarly, changes to the validation fields for a custom field will not automatically result in changes to affected objects, but may require changes the next time those objects are edited in order to bring them into compliance with the updated validation rules.

If a custom field's `default` value is changed, newly created objects will use the new default, but existing objects that were set to the previous default value will *not* automatically be updated.
```